docs/templates-recipes.md

```diff
@@ -0,0 +1,110 @@
+# Templates recipes
+
+In this document you can find recipes for templates that you might want to copy and reuse them for your need to speed up template development. Feel free to create a Pull Request with your recipe.
+
+## Prerequisites
+
+You must be familiar with [Nunjucks](https://mozilla.github.io/nunjucks/) templating engine and [how it is supported in the generator](./authoring.md).
+
+## Recipes
+
+- [Using Schema objects to generate types for a given programming language](#using-schema-objects-to-generate-types-for-a-given-programming-language)
+- [Displaying separate lists for different channels operations](#displaying-separate-lists-for-different-channels-operations)
+
+### Using Schema objects to generate types for a given programming language
+
+* Java example where `$$schema$$.java` file looks like this:
+    ```java
+    package com.async.generated;
+
+    public class {{schema.uid()}} {
+        
+    {%- for propertyName, property in schema.properties() %}
+        public {{property.type() | toJavaType}} {{propertyName}};
+    {%- endfor %}
+        public {{schema.uid()}}(
+        {%- set counter = 0 %}
+        {%- for propertyName, property in schema.properties() %}
+            {%- if property.required()%}
+                {%- if counter == schema.properties().length %}
+                    {{ property.type() | toJavaType }} {{ propertyName }}
+                {%- else %}
+                    {{ property.type() | toJavaType }} {{ propertyName }},
+                {%- endif %}
+            {%- set counter = counter +1 %}
+            {%- endif %}
+        {%- endfor %}) {
+        {%- for propertyName, property in schema.properties() %}
+            {%- if property.required()%}
+                this.{{ propertyName }}={{ propertyName }};
+            {%- endif %}
+        {%- endfor %}
+        }
+    }
+    ```
+    The `toJavaType` filter:
+    ```js
+    Nunjucks.addFilter('toJavaType', jsonSchemaType => {
+    switch (jsonSchemaType.toLowerCase()) {
+        case 'string':
+        return 'String';
+        case 'integer':
+        return  'Integer'
+        case 'number':
+        return 'Double';
+        case 'boolean':
+        return 'boolean';
+    }
```

Comment:
```suggestion
        switch (jsonSchemaType.toLowerCase()) {
            case 'string':
                return 'String';
            case 'integer':
                return  'Integer'
            case 'number':
                return 'Double';
            case 'boolean':
                return 'boolean';
        }
```
Indentation was missing 🤔

Reply:
you sure, lgtm in the preview

Reply:
@derberg this is still the case if you preview the file 😄 Or at least for me 🤔 ?

Reply:
@jonaslagoni it all depends what you expect 😆 I see this and I expect this 😆 
<img width="908" alt="Screenshot 2020-02-28 at 21 22 34" src="https://user-images.githubusercontent.com/6995927/75584666-860c2000-5a70-11ea-8b9e-604f108aa822.png">

Reply:
I just wanted the switch statement to have indentation as well :smile: